plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMetadataFileOperations.java

```diff
@@ -384,10 +385,41 @@ public void testPredicateWithVarcharCastToDate()
         assertUpdate("DROP TABLE test_varchar_as_date_predicate");
     }
 
+    @Test
+    public void testRemoveOrphanFiles()
+            throws Exception
+    {
+        String tableName = "test_deleting_orphan_files_unnecessary_files" + randomTableSuffix();
+        Session sessionWithShortRetentionUnlocked = Session.builder(getSession())
+            .setCatalogSessionProperty("iceberg", "expire_snapshots_min_retention", "0s")
+            .setCatalogSessionProperty("iceberg", "remove_orphan_files_min_retention", "0s")
+            .build();
+        assertUpdate("CREATE TABLE " + tableName + " (key varchar, value integer)");
+        assertUpdate("INSERT INTO " + tableName + " VALUES ('one', 1)", 1);
+        assertUpdate("INSERT INTO " + tableName + " VALUES ('two', 2), ('three', 3)", 2);
+        assertUpdate("DELETE FROM " + tableName + " WHERE key = 'two'", 1);
+
+        assertFileSystemAccesses(
+                sessionWithShortRetentionUnlocked,
+                "ALTER TABLE " + tableName + " EXECUTE REMOVE_ORPHAN_FILES (retention_threshold => '0s')",
+                ImmutableMultiset.builder()
+                        .add(new FileOperation(METADATA_JSON, INPUT_FILE_NEW_STREAM))
+                        .addCopies(new FileOperation(SNAPSHOT, INPUT_FILE_GET_LENGTH), 4)
+                        .addCopies(new FileOperation(SNAPSHOT, INPUT_FILE_NEW_STREAM), 4)
+                        .addCopies(new FileOperation(MANIFEST, INPUT_FILE_GET_LENGTH), 6)
+                        .addCopies(new FileOperation(MANIFEST, INPUT_FILE_NEW_STREAM), 6)
```

Comment:
Looks like we're still reading each manifest twice, I'll take a look at that tomorrow.

Reply:
I think that first every manifest is read here:
```
for (ManifestFile manifest : snapshot.allManifests(table.io())) 
```
and then again here:
```
ManifestFiles.read(manifest, table.io());
ManifestFiles.readDeleteManifest(manifest, table.io(), table.specs());
```

but this is still huge improvement as previously these were the numbers:
```
.addCopies(new FileOperation(SNAPSHOT, INPUT_FILE_GET_LENGTH), 4)
.addCopies(new FileOperation(SNAPSHOT, INPUT_FILE_NEW_STREAM), 4)
.addCopies(new FileOperation(MANIFEST, INPUT_FILE_GET_LENGTH), 24)
.addCopies(new FileOperation(MANIFEST, INPUT_FILE_NEW_STREAM), 24)
```

Reply:
Looks like it's part of how the ManifestReader works. The factory  `ManifestFiles.read` opens the file once to read the metadata, and then calling `iterator` opens the file again. I don't think there's much we can do about that